x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunnerTests.java

```diff
@@ -0,0 +1,231 @@
+                "}",
+                XContentType.JSON)
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+        ).get();
+
+        logger.info("Status: " + indexRequest.status().getStatus());
+
+        SearchResponse sourceSearchResponse = client().search(
+            new SearchRequest(sourceIndex)
+                .source(SearchSourceBuilder.searchSource()
+                    .query(QueryBuilders.matchAllQuery()))).get();
+
+        assertThat(sourceSearchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Map<String, Object> sourceDocMap = sourceSearchResponse.getHits().getAt(0).getSourceAsMap();
+        assertNotNull(sourceDocMap);
+        assertThat(sourceDocMap.get("field1"), is(equalTo("value1")));
+        assertThat(sourceDocMap.get("field2"), is(equalTo(2)));
+        assertThat(sourceDocMap.get("field3"), is(equalTo("ignored")));
+        assertThat(sourceDocMap.get("field4"), is(equalTo("ignored")));
+        assertThat(sourceDocMap.get("field5"), is(equalTo("value5")));
+
+        logger.info("Created Doc");
+        ClusterService clusterService = getInstanceFromNode(ClusterService.class);
+        IndexNameExpressionResolver resolver = getInstanceFromNode(IndexNameExpressionResolver.class);
+
+        final long createTime = randomNonNegativeLong();
+
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        List<String> enrichFields = new ArrayList<>();
+        enrichFields.add("field2");
+        enrichFields.add("field5");
+        EnrichPolicy policy = new EnrichPolicy(EnrichPolicy.EXACT_MATCH_TYPE, null, sourceIndex, "field1", enrichFields, "");
+        String policyName = "test1";
+
+        ActionListener<PolicyExecutionResult> listener = new ActionListener<PolicyExecutionResult>() {
+            @Override
+            public void onResponse(PolicyExecutionResult policyExecutionResult) {
+                logger.info("Run complete");
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.warn("Run failed");
+                exception.set(e);
+                latch.countDown();
+            }
+        };
+
+        EnrichPolicyRunner enrichPolicyRunner =
+            new EnrichPolicyRunner(policyName, policy, listener, clusterService, client(), resolver, () -> createTime);
+
+        logger.info("Starting policy run");
+
+        enrichPolicyRunner.run();
+
+        latch.await();
+        if (exception.get() != null) {
+            throw exception.get();
+        }
+
+        GetIndexResponse enrichIndex = client().admin().indices()
+            .getIndex(new GetIndexRequest().indices(".enrich-test1")).get();
+        assertThat(enrichIndex.getIndices().length, equalTo(1));
+        assertThat(enrichIndex.getIndices()[0], equalTo(".enrich-test1-" + createTime));
```

Comment:
maybe also assert that the enrich key field has been defined with the correct attributes set, _source is disabled, dynamic field mapping is set to false and auto expand replicas are enabled?